docusaurus/src/remark/docsHeaderDecoration.js

```diff
@@ -0,0 +1,135 @@
+const fetch = require("node-fetch");
+const visit = require("unist-util-visit");
+
+const registry_url =
+  "https://connectors.airbyte.com/files/generated_reports/connector_registry_report.json";
+let registry = {};
+let loading = false;
+
+const plugin = () => {
+  const transformer = async (ast, vfile) => {
+    if (!isDocsPage(vfile)) return;
+
+    await fetchCatalog();
+
+    const pathParts = vfile.path.split("/");
+    const connectorName = pathParts.pop().split(".")[0];
+    const connectorType = pathParts.pop();
+    const dockerRepository = `airbyte/${connectorType.replace(
+      /s$/,
+      ""
+    )}-${connectorName}`;
+    const registryEntry = registry.find(
+      (r) => r.dockerRepository_oss === dockerRepository
+    );
+
+    if (!registryEntry) return;
+
+    visit(ast, "heading", (node) => {
+      let headerFound = false;
+      if (node.depth === 1 && node.children.length === 1 && !headerFound) {
+        headerFound = true;
+        const originalTitle = node.children[0].value;
+        const originalId = node.data.hProperties.id;
+
+        node.type = "html";
+        node.children = undefined;
+        node.value = buildConnectorHTMLContent(
+          registryEntry,
+          originalTitle,
+          originalId
+        );
+      }
+    });
+  };
+  return transformer;
+};
+
+const buildConnectorHTMLContent = (
+  registryEntry,
+  originalTitle,
+  originalId
+) => {
+  const htmlContent = `
+  <h1 id="${originalId}">${originalTitle}</h1>
+  <small>
+    <table>
+      <tbody>
+        <tr>
+          <td rowSpan="4">
+            <img style="max-height: 75px" src="${registryEntry.iconUrl_oss}" />
+          </td>
+          <td>Support Level: </td>
+          <td><strong>${capitalizeFirstLetter(
+            registryEntry.supportLevel_oss
+          )}</strong></td>
+        </tr>
+        <tr>
+          <td>Definition Id: </td>
+          <td>${registryEntry.definitionId}</td>
+        </tr>
+        <tr>
+          <td>Latest Version: </td>
+          <td>${registryEntry.dockerImageTag_oss}</td>
+        </tr>
+        <tr>
+          <td>Availability:</td>
+          <td>Airbyte Cloud: ${
+            registryEntry.is_cloud ? "✅" : "❌"
+          }  |  Airbyte OSS: ${registryEntry.is_oss ? "✅" : "❌"}</td>
+        </tr>
+      </tbody>
+    </table>
+  </small>
+  <br />
+`;
+
+  return htmlContent;
+};
+
+const isDocsPage = (vfile) => {
+  if (
+    !vfile.path.includes("integrations/sources") &&
+    !vfile.path.includes("integrations/destinations")
+  ) {
+    return false;
+  }
+
+  if (vfile.path.includes("-migrations.md")) {
+    return false;
+  }
+
+  return true;
+};
+
+/*
+This methods makes registry a singleton available to all callers of this plugin.
+Only one download will happen for the first callers and all others will use the cached version.
+*/
+const fetchCatalog = async () => {
```

Comment:
nit: Would love if this was hoisted to the top so its right next to the global registry variable

Reply:
This whole method seems to be not necessary to me? Can't we simply do inside the `plugin`, but outside the `transform` function a single `fetch` call assign that returning to a local variable and use that everywhere?

i.e. along the lines of

```js
const plugin = () => {
  const registryFetch = fetch("...");
  const transformer = async () => {
    const registry = await registryFetch();

  }
  return transformer;
```

I don't see why we'd need any loading state or similar here?

Reply:
The plugin gets called by every page, so I wanted a global way to block downloading the file more than once, and they all get invoked at the same time

Reply:
In this case it should be fine pulling it just out to the top level. Modules are singletons by default in node (why the registry variable above works).